### ISSUE TITLE
Add readthedocs documentation to Scribe-Data

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Data#57]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Data/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Data's [Code of Conduct](https://github.com/scribe-org/Scribe-Data/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue would be to add in [readthedocs](https://about.readthedocs.com/?ref=readthedocs.com) for Scribe-Data. Goal would be to have all functions and classes automatically documented. 

- Note that this issue is linked to #56 😊

### Contribution

Happy to support with this! We should discuss implementation beforehand, but I've used readthedocs extensively for other projects and am happy to help with setup or add parts to the docs.

**Comment by shashank-iitbhu** [comment]
assign @andrewtavis

**Comment by andrewtavis** [comment]
Thank you @shashank-iitbhu!

**Comment by m-charlton** [comment]
Apologies for the late reply. Started to look at this late last year but, other things got in the way.
Will gather my notes and lay out a plan of action

**Comment by andrewtavis** [comment]
Great to hear from you, @m-charlton! Plan is to get the baseline for this up for @shashank-iitbhu later today. We can talk about this here in the issue and also check into the work for the doc strings in #56 :)

**Comment by andrewtavis** [comment]
Ok all 😊 We now have the base docs available at [scribe-data.readthedocs.io](https://scribe-data.readthedocs.io/), and the GitHub integration is working fine on commits to the repo 🙌 In setting it up, for some reason the modules weren't being loaded in though. I'm wondering if maybe we didn't have the current `__init__.py` structure when the last release was made? Not sure. My assumption is that the following in [utils.rst](https://github.com/scribe-org/Scribe-Data/blob/main/docs/source/utils.rst) should work:

```
utils
=====

The :py:mod:`utils` module provides utility functions for data extraction, formatting and loading.

.. automodule:: scribe_data.utils
    :members:
    :private-members:
```

What we're getting though is:

```
WARNING: autodoc: failed to import module 'utils' from module 'scribe_data'; the following exception was raised:
No module named 'scribe_data.utils'
```

Similarly for [checkquery](https://github.com/scribe-org/Scribe-Data/blob/main/docs/source/checkquery.rst). Let's share some thoughts, and then I think we should be good to try out `automodule` with other files. Note that we are using `numpydoc` as of now, so for #56 some of the docstrings might need to be changed. If others have a suggestion on other doc styles we can definitely make a change though!

**Comment by andrewtavis** [comment]
CC @wkyoshida :) Another thing to consider is that it'd be good to set it up with a `Edit on GitHub` link or something like that in the top right rather than the `View source` (not too much value there I'd say). This can be a separate issue though if we want 😊

**Comment by andrewtavis** [comment]
I would say in general a couple more links to GitHub in there would be good. Maybe something in the left sidebar as well?

**Comment by andrewtavis** [comment]
Also of note, here's an awesome list for readthedocs: [github.com/readthedocs-examples/awesome-read-the-docs](https://github.com/readthedocs-examples/awesome-read-the-docs). I think that for this issue we can focus on finalizing getting all the function docstrings in here along with #56, and then we can make a few more issues for customizing it a bit with things like a logo in the sidebar, etc.

**Comment by andrewtavis** [comment]
Very long day of work on this to get it ready for docs and the next steps for GSoC, but we're basically there 😊 One minor release and two patches to get the docs working in a day 🚀

Where we're at for the documentation can be seen at [scribe-data.readthedocs.io](https://scribe-data.readthedocs.io/en/latest/). I did some basic organizing, and autodoc/automodule is working. We can now apply this over the rest of the project and then this issue will be finished :)

Of note is that I got rid of the conda environment and we're now using Python venv. Was necessary to do a release so that tensorflow wouldn't needlessly be installed when we install Scribe-Data for the docs. To get the venv up and running we can do the following:

```bash
# In the root of the project:
python3 -m venv venv  # make an environment venv
pip install --upgrade pip  # make sure that pip is at the latest version
pip install -r requirements.txt  # install dependencies
```

If you're using VS Code, then it should prompt you to select this as the default environment. From there restart the terminal session in VS Code to then see that you're in the venv.

To get the docs up and running:

```bash
cd docs
make html
```

You can then open `index.html` within `docs/build/html` to check the local version of the documentation.

Let's discuss all of this in the sync! 😊

**Comment by mhmohona** [comment]
I want to help in this issue! 
Is there anything I can do regrading writing documentation or setting things up?

**Comment by andrewtavis** [comment]
Hey @shashank-iitbhu 👋 Do you want to send the autodoc/automodule calls to finalize the documentation and we can close this?

**Comment by andrewtavis** [comment]
```
.. automodule:: scribe_data.utils
    :members:
    :private-members:
```

**Comment by shashank-iitbhu** [comment]
> Hey @shashank-iitbhu 👋 Do you want to send the autodoc/automodule calls to finalize the documentation and we can close this?

Looking into this!

**Comment by shashank-iitbhu** [comment]
In `conf.py` the `sys.path` is this:
```python
sys.path.insert(0, os.path.abspath(".."))
``` 
don't you think, it should be :
```python
sys.path.insert(0, os.path.abspath("../../src"))
``` 

currently getting this error, when i do `cd docs && make html` :
<img width="390" alt="Screenshot 2024-03-10 at 5 08 14 PM" src="https://github.com/scribe-org/Scribe-Data/assets/134289475/3c5bab60-fa3d-4ade-a02c-3dc31cc2abdd">

After updating the path to `../../src`
i was able to generate docs for all the functions in `checkquery.py` and `utils.py`.

<img width="1469" alt="Screenshot 2024-03-10 at 3 59 28 PM" src="https://github.com/scribe-org/Scribe-Data/assets/134289475/5ea29861-5932-4868-93e8-923b7a070650">

**Comment by shashank-iitbhu** [comment]
cc @andrewtavis

**Comment by andrewtavis** [comment]
Closed via 18046e3 🥳 Definitely will need to be updated as we go, but this we finally have a baseline for the documentation for Scribe-Data! 🙌 Thanks all so much for your contributions!
